comictaggerlib/ui/talkeruigenerator.py

```diff
@@ -0,0 +1,233 @@
+    # Create a label
+    layout.addWidget(lbl, layout.rowCount() + 1, 0)
+    widget = QtWidgets.QLineEdit()
+    widget.setObjectName(option.internal_name)
+    widget.setToolTip(option.help)
+    layout.addWidget(widget, layout.rowCount() - 1, 1)
+    # Special case for api_key, make a test button
+    if option.internal_name.endswith("api_key"):
+        btn = QtWidgets.QPushButton("Test Key")
+        layout.addWidget(btn, layout.rowCount() - 1, 2)
+
+    if option.internal_name.endswith("url"):
+        btn = QtWidgets.QPushButton("Test URL")
+        layout.addWidget(btn, layout.rowCount() - 1, 2)
+
+    return layout, widget, btn
+
+
+def settings_to_talker_form(sources: dict[str, QtWidgets.QWidget], config: settngs.Config[settngs.Namespace]):
+    for talker in sources.items():
+        for name, widget in talker[1]["widgets"].items():
+            value = getattr(config[0], name)
+            value_type = type(value)
+            try:
+                if value_type is str:
+                    # Special case for general dropdown box
+                    if name == "talker_source":
+                        widget.setCurrentIndex(widget.findData(config[0].talker_source))
+                    else:
+                        widget.setText(value)
+                if value_type is int or value_type is float:
+                    widget.setValue(value)
+                if value_type is bool:
+                    widget.setChecked(value)
+            except Exception:
+                logger.debug("Failed to set value of %s", name)
+
+
+def generate_source_option_tabs(
+    tabs: QtWidgets.QTabWidget,
+    config: settngs.Config[settngs.Namespace],
+    talkers: dict[str, ComicTalker],
+) -> dict[str, QtWidgets.QWidget]:
+    """
+    Generate GUI tabs and settings for talkers
+    """
+
+    sources: dict = {}
+    # Use a dict to make a var name from var
+    source_info = {}
+
+    # Add General tab
+    source_info["general"] = {"tab": QtWidgets.QWidget(), "widgets": {}}
+    general_layout = QtWidgets.QGridLayout()
+    lbl_info = QtWidgets.QLabel("Information Source:")
+    cbx_info = QtWidgets.QComboBox()
+    general_layout.addWidget(lbl_info, 0, 0, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Maximum)
+    general_layout.addWidget(cbx_info, 0, 1, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Maximum)
+    vspacer = QtWidgets.QSpacerItem(20, 40, QtWidgets.QSizePolicy.Minimum, QtWidgets.QSizePolicy.Expanding)
+    general_layout.addItem(vspacer, 1, 0)
+    source_info["general"]["widgets"]["talker_source"] = cbx_info
+
+    source_info["general"]["tab"].setLayout(general_layout)
+    tabs.addTab(source_info["general"]["tab"], "General")
+
+    # Add source sub tabs to Comic Sources tab
+    for talker_id, talker_obj in talkers.items():
+        # Add source to general tab dropdown list
+        source_info["general"]["widgets"]["talker_source"].addItem(talker_obj.name, talker_id)
```

Comment:
I've done the `cbx_info` change and also added it to the sources widget list. As for the finddata, afaik this is the best way to set the current item in the combo box? Even if there is only one talker now, there won't be so I think it makes sense to put this in now rather than say, with a second talker addition.

Reply:
By find I just meant iterating through the data in order to find the single combobox.